public/restaurants.js

```diff
@@ -0,0 +1,158 @@
+
+// gets restaurant id from pathname
+
+let path = window.location.pathname
+
+let pathArray = path.split('/') // gets the id number which is the last part of the file path (after /)
+
+let id = pathArray.pop()
+
+// other variables from HTML
+
+let comments = document.getElementById('comments')
+
+let restaurantInfo = document.getElementById('restaurant-info')
+
+let button = document.getElementById('button')
+let message = document.getElementById('feedback')
+
+// variables for storing comments in local storage
+
+let savedComments = []
+let storedNotes = []
+
+// function to add inputted comments to comment space and call function to store them in local storage
+
+button.addEventListener('click', function () {
+	let transferredComments = localStorage.getItem(id + '-notes') || []
+	if (typeof transferredComments === 'string') {
+		savedComments = transferredComments.split(',')
+		savedComments.push(message.value)
+		populateStorage(id)		
+	}
+
+	if (typeof transferredComments === 'object') {
+		savedComments.push(message.value)
+		populateStorage(id)
+	}
+	comments.innerHTML += `<li style='padding-top: 1vh; list-style-type: none'>"${message.value}"</li>`
+	message.value = ""
+})
+
+// function to store feedback 
+
+function populateStorage(pageId) {
+	localStorage.setItem(pageId + '-notes', savedComments.toString())
+}
+
+// function to repopulate comments section
+
+function retrieveSavedComments() {
+	let retrievedComments = localStorage.getItem(id + '-notes')
+	if (typeof retrievedComments === 'string') {		
+		let storedNotes = retrievedComments.split(',')
+		storedNotes.forEach((storedComment) => {
+			comments.innerHTML += `<li style='padding-top: 1vh; list-style-type: none'>"${storedComment}"</li>`
+		})
+	}
+}
+
+// function to retrieve restaurant info from my api
+
+async function getMoreRestoInfo() {
+
+	let restaurant = await fetch('/api/indexRestaurants.json')
+		.then((response) => {
+			return response.json()
+		}).then((jsonArr) => {
+			return jsonArr
+		})
+
+	// iterates over index file to find the specific restaurant
+
+	let resto = restaurant.find((obj) => obj.id === id)
+
+	// initializing variables for info to be added to sidebar on restaurant page
+
+	let name = resto.name
+	let address = resto.address
+	let phone = resto.phone
+	let website = resto.website
+	let hours = resto.hours
+	let notes = resto.notes
+
+	// checks for missing or aberrant information 
+
+	if (!resto.hasOwnProperty('hours')) { 
+		hours = 'No hours available'
+	}
+
+	if (!resto.hasOwnProperty('website')) { 
+		website = ''
+	}
+	if (phone.length <= 8) {
+		phone = "802-" + phone
+	}
+	// 
+	// adds restaurant info to restaurant page
+
+	restaurantInfo.innerHTML = `<li><p style='font-size: 3vh; text-decoration: underline'>${name}</p></li><li><p>${address}</p></li><li><p>${phone}</p></li><li><a href='${website}' target="_blank">${website}</a></li><li><p>${hours}</p></li>`
```

Comment:
This works, but is a little messy and hard to read. You could use something like:
```js
// create list items
let nameLI = document.createElement('li');
let addressLI = document.createElement('li');
let phoneLI = document.createElement('li');
let websiteLI = document.createElement('li');
let hoursLI = document.createElement('li');

// create anchor for link
let websiteAnchor = document.createElement('a');

// set text content of each item
nameLI.textContent = name;
addressLI.textContent = address;
phoneLI.textContent = phone;
websiteAnchor.textContent = website;
hoursLI.textContent = hours

// set attribute to anchor, append anchor to corresponding LI
websiteAnchor.setAttribute('href', website);
websiteLI.appendChild(websiteAnchor);

// append all list items to restaurantInfo ul
restaurantInfo.append(nameLI, addressLI, phoneLI, websiteLI, hoursLI);
```

It's more lines, but a bit more clear what exactly you are doing at a glance.